### PR TITLE
(TWILL-112) Improve performance of ApplicationBundler

### PR DESCRIPTION
**Pull request by chtyim**
- Not to scan a directory again if the directory was added to the bundle jar that is building in progress already

### CODE REVIEW

File: twill-core/src/main/java/org/apache/twill/internal/ApplicationBundler.java [hunk 1]
```diff
@@ -175,6 +175,9 @@ public String apply(Class<?> input) {
     if (classLoader == null) {
       classLoader = getClass().getClassLoader();
     }
+
+    // Record the set of classpath URL that are already added to the jar
+    final Set<URL> seensClassPaths = Sets.newHashSet();
```
Comment (ghelmling) [inline twill-core/src/main/java/org/apache/twill/internal/ApplicationBundler.java]: nit: "seensClassPaths" should be "seenClassPaths" or "addedClassPaths"

**Comment by ghelmling** [comment]
+1 with minor nit on naming.

File: twill-core/src/main/java/org/apache/twill/internal/ApplicationBundler.java (hunk 1 above)
Comment (chtyim) [inline twill-core/src/main/java/org/apache/twill/internal/ApplicationBundler.java]: Good catch. Fixed.

**Comment by ghelmling** [comment]
+1

**Pull request closed without merging**